pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -197,25 +196,16 @@ var _ = g.Describe("[Feature:Machines] Autoscaler should", func() {
 			g.Skip(fmt.Sprintf("MachineSet %s is owned by a machineDeployment. Please run tests against machineDeployment instead", targetMachineSet.Name))
 		}
 
-		g.By(fmt.Sprintf("Create ClusterAutoscaler and MachineAutoscaler objects. Targeting machineSet %s", targetMachineSet.Name))
-		initialNumberOfReplicas := *targetMachineSet.Spec.Replicas
+		g.By("Create ClusterAutoscaler object")
 		clusterAutoscaler := clusterAutoscalerResource()
+		err = client.Create(context.TODO(), clusterAutoscaler)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		initialNumberOfReplicas := *targetMachineSet.Spec.Replicas
```

Comment:
This is a general comment for the PR: we should choose to deref safely using `pointer.Int32PtrDerefOr()`.